### PR TITLE
feat(test) - [WD-37811] Follow-up test for symlink resolution

### PR DESCRIPTION
**Pull request by Kxiru**
## Done

Creates a spec test in the utils folder to call the resolveSymlnkTarget function with different values and verify it’s output.

Tests:
- Relative paths
- Root paths
- Paths to directories
- Paths to files
- Paths to other symlinks

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @Kxiru or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](https://github.com/canonical/lxd-ui/blob/main/CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - [List the steps to QA the new feature(s) or prove that a bug has been resolved]

## Screenshots

N/A

### CODE REVIEW

**Review by tarek-hanna — DISMISSED** [review]
LGTM .. and Copilot didn't complain either :D

 I wonder if it would make sense to add cases for empty string and null input, to validate error handling or possible edge cases, and also  2 relative paths (as input), since all cases here have at least 1 abs path.

**Comment by Kxiru** [comment]
> LGTM .. and Copilot didn't complain either :D
> 
> I wonder if it would make sense to add cases for empty string and null input, to validate error handling or possible edge cases, and also 2 relative paths (as input), since all cases here have at least 1 abs path.

I think that is a valid concern, I will add these now.

File: src/util/instances.spec.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  describe("two relative paths", () => {
+    it("resolves a sibling target with a relative currentPath", () => {
+      expect(resolveSymlinkTarget("bar", "home/user/file")).toBe(
+        "/home/user/bar",
+      );
+    });
+
+    it("resolves a target with .. and a relative currentPath", () => {
+      expect(resolveSymlinkTarget("../other", "home/user/file")).toBe(
+        "/home/other",
```
Comment (tarek-hanna) [inline src/util/instances.spec.ts]: not sure if `home` being **relative** and suddenly becoming **absolute** is correct. i would expect it to resolve to something like `/root/.../home/user/other`

File: src/util/instances.spec.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    it("throws when currentPath is null", () => {
+      expect(() =>
+        resolveSymlinkTarget("bar", null as unknown as string),
+      ).toThrow();
+    });
+  });
+
+  describe("two relative paths", () => {
+    it("resolves a sibling target with a relative currentPath", () => {
+      expect(resolveSymlinkTarget("bar", "home/user/file")).toBe(
```
Comment (tarek-hanna) [inline src/util/instances.spec.ts]: same here. it should first get the absolute path for home wherever it  lives and then handles the relative path  in respect to bar, but again how does that work if i flip the input `expect(resolveSymlinkTarget( "home/user/file", "bar")).toBe( ??)`  >>> `/bar/home/user/file` does the input order matter here ?

Comment (tarek-hanna) [reply src/util/instances.spec.ts]: i see `target` goes first and then `current path` so it might be easier to use if you enforce  the current path to always be absolute?  and check for `starts.with('/')` WDYT? this way the relative  path in the second input will say it is invalid input and should be absolute... again i dont know the exact use cases you need this symlink method for, so feel free to correct me or discard this :D

**Comment by Kxiru** [comment]
@tarek-hanna Fair enough, we're testing undefined behaviour in this relativePaths block as currentPath is never relative. I will remove this test block entirely,  but I believe the Edge cases block is still relevant.

**Comment by tarek-hanna** [comment]
> @tarek-hanna Fair enough, we're testing undefined behaviour in this relativePaths block as currentPath is never relative. I will remove this test block entirely, but I believe the Edge cases block is still relevant.

so basically what you are saying here is that we should enforce validation on the targetPath being absolute since "it can never be relative"?

**Review by tarek-hanna — APPROVED** [review]
Just to keep things in scope and unblock you. PR is approved. feel free to close/merge this one and have a follow up later with the extra validations (if needed)

**Comment by Kxiru** [comment]
> so basically what you are saying here is that we should enforce validation on the targetPath being absolute since "it can never be relative"?

I'm saying that the currentPath is never relative, so it was unnecessary to test current paths that _are_. I'm not sure what you meant when you brought up the target paths though.

Thank you for unblocking!
